app/src/org/commcare/activities/EntitySelectActivity.java

```diff
@@ -596,6 +596,8 @@ public void onActivityResultSessionSafe(int requestCode, int resultCode, Intent
                         }
                         AudioController.INSTANCE.releaseCurrentMediaEntity();
                     }
+                    // Reload entities in case some are no longer available after the sync
+                    loadEntities();

```

Comment:
Not necessarily, we might want to avoid reloading the entities every time `onResume()` is called. Also, `loadEntities()` is already being called in [`refreshView()`](https://github.com/dimagi/commcare-android/blob/1194304aa837e2e412591461fa7adb3207797a55/app/src/org/commcare/activities/EntitySelectActivity.java#L436) when the adapter is `null`.